src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -79,7 +73,7 @@
 /**
  *
  */
-public class InternalEngine implements Engine {
+public class InternalEngine extends AbstractEngine {
 
     protected final ESLogger logger;
```

Comment:
we don't this any more right? it's in AbstractEngine. Why not move shardId as well? seems useful for sub classes.